packages/wrangler/src/deploy/index.ts

```diff
@@ -281,22 +284,15 @@ export async function deployHandler(
 	}
 
 	if (
-		(args.legacyAssets ||
-			config.legacy_assets ||
-			args.experimentalAssets ||
-			config.experimental_assets) &&
+		(args.legacyAssets || config.legacy_assets) &&
 		(args.site || config.site)
 	) {
 		throw new UserError(
-			"Cannot use Assets and Workers Sites in the same Worker."
+			"Cannot use Legacy Assets and Workers Sites in the same Worker."
 		);
 	}
 
-	if (args.assets) {
```

Comment:
removing this because duplicate of what we have on line `249` 👇 

```
if (args.assets) {
		logger.warn(
			`The --assets argument is experimental. We are going to be changing the behavior of this experimental command after August 15th.\n` +
				`Releases of wrangler after this date will no longer support current functionality.\n` +
				`Please shift to the --legacy-assets command to preserve the current functionality.`
		);
	}
```